### ISSUE TITLE
outjars doesn't work for output directories in Gradle task

### ISSUE THREAD

**Issue by guillerodriguez** [Guardsquare/proguard#106]
I am using the Gradle task to obfuscate/minify a regular Java application. I want obfuscated classes to be written to a directory (not to an output jar), so I use this in my build.gradle file:

```
task ('proguard', type: proguard.gradle.ProGuardTask) {
	dependsOn classes

	injars sourceSets.main.output
	outjars "${buildDir}/proguard"
	[...]
```

If the directory does not exist (e.g. on a first run) this works as expected. The directory is created and all classes are correctly written.

If the directory already exists (e.g. on a second run without doing a `clean` first) this fails:

```
FAILURE: Build failed with an exception.

* What went wrong:
A problem was found with the configuration of task ':myapp:proguard' (type 'ProGuardTask').
> Cannot write to file '/Users/grodriguez/myapp/gradle-build/proguard' specified for property 'outJarFileCollection' as it is a directory.
org.gradle.internal.execution.WorkValidationException: A problem was found with the configuration of task ':myapp:proguard' (type 'ProGuardTask').
        at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:71)
        [...]
Caused by: org.gradle.api.InvalidUserDataException: Cannot write to file '/Users/grodriguez/myapp/gradle-build/proguard' specified for property 'outJarFileCollection' as it is a directory.
        ... 39 more
```

This is happening with ProGuard 7.0.1.

**Comment by guillerodriguez** [comment]
Probably cause by the `@OutputFiles` annotation: https://github.com/Guardsquare/proguard/blob/679425e5d87834e3f7cd1cbe09f120a410c67a66/gradle-plugin/src/proguard/gradle/ProGuardTask.java#L77

This causes Gradle to complain if the file exists but is a directory..

**Comment by mrjameshamilton** [comment]
Hi @guillerodriguez! Thanks for the report.

ProGuard itself (not the Gradle task) will create the directories that you specify in `-outjars`. The task behaviour here is a little inconsistent with using ProGuard in standalone mode, since you wouldn't get an error in that case if they directory already exists.

I've tagged this with `enhancement`; there are several problems with the current Gradle integration. We're currently thinking about how we can improve the integration.

You could create a `clean` task to remove the directory before running the proguard task e.g.

```groovy
task cleanOutputDir(type: Delete) {
    delete "${buildDir}/proguard"
}

task ('proguard', type: proguard.gradle.ProGuardTask) {
        dependsOn cleanOutputDir
	dependsOn classes

	injars sourceSets.main.output
	outjars "${buildDir}/proguard"
	[...]
```

**Comment by mrjameshamilton** [comment]
Fixed as part of #139

**Comment by hakanai** [comment]
This issue is still present in v7.2.2.

I agree that the `@OutputFiles` annotation is probably the issue. Having a separate property which has `@OutputDirectory` instead would probably be the fix?

The workaround of deleting the directory in a custom `onlyIf` block which we were using in v6.2.2 still seems to work in the new version.
